loopchain/store/key_value_store_dict.py

```diff
@@ -0,0 +1,149 @@
+# limitations under the License.
+"""KeyValueStoreDict classes are components for development"""
+
+import functools
+
+from loopchain.store.key_value_store import KeyValueStoreError
+from loopchain.store.key_value_store import KeyValueStoreWriteBatch, KeyValueStoreCancelableWriteBatch, KeyValueStore
+from loopchain.store.key_value_store import _validate_args_bytes, _validate_args_bytes_without_first
+
+
+def _error_convert(func):
+    @functools.wraps(func)
+    def _wrapper(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except Exception as e:
+            raise KeyValueStoreError(e)
+
+    return _wrapper
+
+
+class _KeyValueStoreWriteBatchDict(KeyValueStoreWriteBatch):
+    def __init__(self, store_items: dict):
+        self._store_items = store_items
+        self._batch_items = dict()
+
+    @_validate_args_bytes_without_first
+    @_error_convert
+    def put(self, key: bytes, value: bytes):
+        self._batch_items[key] = value
+
+    @_validate_args_bytes_without_first
+    @_error_convert
+    def delete(self, key: bytes):
+        self._batch_items[key] = None
+
+    @_error_convert
+    def clear(self):
+        self._batch_items.clear()
+
+    @_error_convert
+    def write(self):
+        for key, value in self._batch_items.items():
+            if value is None:
+                try:
+                    del self._store_items[key]
+                except KeyError:
+                    pass
+            else:
+                self._store_items[key] = value
+
+
+class _KeyValueStoreCancelableWriteBatchDict(KeyValueStoreCancelableWriteBatch):
+    def __init__(self, store: KeyValueStore, store_items: dict):
+        super().__init__(store)
+        self._store_items = store_items
+        self._original_items = dict()
+
+    def _touch(self, key: bytes):
+        if key in self._original_items:
+            return
+
+        try:
+            value = self._store_items[key]
+        except KeyError:
+            value = None
+        self._original_items[key] = value
+
+    def _get_original_touched_item(self):
+        for key, value in self._original_items.items():
+            yield key, value
+
+    def clear(self):
+        super().clear()
+        self._original_items.clear()
+
+    def close(self):
+        self._original_items: dict = None
+
+
+class KeyValueStoreDict(KeyValueStore):
+    """KeyValueStoreDict class is just for development"""
+
+    def __init__(self):
+        self._store_items = dict()
+
+    @_validate_args_bytes_without_first
+    @_error_convert
+    def get(self, key: bytes, default=None, **kwargs) -> bytes:
+        if default is not None:
+            _validate_args_bytes(default)
+
+        try:
+            return self._store_items[key]
+        except KeyError:
+            if default is None:
+                raise KeyError(f"Has no value of key({key}")
+            return default
+
+    @_validate_args_bytes_without_first
```

Comment:
Passing parameter explicitly sync=True raises an exception.

Reply:
I don't know how did you see exception raised.
Can you explain more detail?